### DISCUSSION TITLE
como arrumar esse erro na hora de usar trainamento lora, por favor

### DISCUSSION THREAD

**Discussion by yoshyteru** [nod-ai/SHARK#1240] [General]
Traceback (most recent call last):
  File "gradio\routes.py", line 394, in run_predict
  File "gradio\blocks.py", line 1075, in process_api
  File "gradio\blocks.py", line 884, in call_function
  File "anyio\to_thread.py", line 31, in run_sync
  File "anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
  File "anyio\_backends\_asyncio.py", line 867, in run
  File "apps\stable_diffusion\scripts\train_lora_word.py", line 251, in lora_train
  File "torch\nn\modules\module.py", line 1126, in to
    device, dtype, non_blocking, convert_to_format = torch._C._nn._parse_to(*args, **kwargs)
                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Invalid device string: 'AMD Radeon RX 6600 => vulkan://0'

**Comment by cstueckrath** [comment]
backend implementation for training was not released upstream.
You can help yourself if you run it in Linux (WSL should work, too) on top of master with this guide:
_(A implementação backend para formação não foi libertada a montante.
Pode ajudar-se a si próprio se a executar em Linux (WSL também deve funcionar) em cima do mestre com este guia:)_
https://github.com/nod-ai/SHARK/tree/main/shark/examples/shark_training/stable_diffusion

**Comment by yoshyteru** [comment]
Hello, good evening, my system is windowns 11, what would be the correct way to install x formers, torch 2.0,

I see a lot of limitation in relation to amd and windows for AI ART, this is very discouraging :(  Thanks in advance
Enviado do Email<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: ***@***.***>
Enviado:sexta-feira, 24 de março de 2023 05:57
Para: ***@***.***>
***@***.***>; ***@***.***>
Assunto: Re: [nod-ai/SHARK] como arrumar esse erro na hora de usar trainamento lora, por favor (Discussion #1240)


backend implementation for training was not released upstream.
You can help yourself if you run it in Linux (WSL should work, too) on top of master with this guide:
https://github.com/nod-ai/SHARK/tree/main/shark/examples/shark_training/stable_diffusion
